In [1]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from IPython.display import IFrame
from jupyterthemes import jtplot

In [2]:
# jtplot.style(theme='monokai')

In [3]:
plt.style.use('default')
plt.rc('font',size=14)
plt.rc('figure',figsize=(8,6))
plt.rc('text', usetex=True)


En este notebook me encargo solamente de obtener las masas efectivas isotrópicas. Hay distitntos casos. según [wikipedia](https://en.wikipedia.org/wiki/Effective_mass_(solid-state_physics))
hay una masa efectiva de la densidad de estados:
$$
m^{*}_{density} = \sqrt[3]{g^2 m_x m_y m_z}
$$

donde $g$ es un factor de degeneración que dice la multiplicidad de valles en la zona de brillouin.

o tambien está la masa efectiva segun el modelo de Drude:
$$
m^{*}_{conductivity} = \frac{1}{3}\left[ 
\frac{1}{m_x} +
\frac{1}{m_y} +
\frac{1}{m_z} +
\right]^{-1}
$$
pero, esta última vale er raras ocaciones porque la ley de drude depende tambien de la movilidad de los portadores.

In [4]:
datavol = pd.read_csv('datavol.dat', sep=' ')

In [7]:
# vopt = datavol['Vol'][datavol['evsk4'].dimin()]
vopt=datavol['Vol'][datavol['Evsk4'].idxmin()]
#print('El volumen de minima energía es: {}'.format(vopt))

El volumen de minima energía es {{vopt}}

# Estructura de bandas con calculo bueno

In [5]:
IFrame('make_valenceband_wall.pdf',width=1000, height=500)


In [21]:
databands = pd.read_csv('more_useful_data.txt',sep='#') 

In [22]:
databands

Vol mass   value  bandno    spin                from     fname  \
0    60.12  m_h  -0.588      23    (up)  [0.50, 0.50, 0.50]       (A)   
1    60.12  m_h  -0.605      23    (up)  [0.50, 0.50, 0.50]       (A)   
2    60.12  m_h  -0.588      24    (up)  [0.50, 0.50, 0.50]       (A)   
3    60.12  m_h  -0.605      24    (up)  [0.50, 0.50, 0.50]       (A)   
4    60.12  m_e   0.557      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
5    60.12  m_e   1.946      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
6    60.12  m_e   0.557      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
7    60.12  m_e  16.797      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
8    62.04  m_h  -0.628      23    (up)  [0.50, 0.50, 0.50]       (A)   
9    62.04  m_h  -0.646      23    (up)  [0.50, 0.50, 0.50]       (A)   
10   62.04  m_h  -0.628      24    (up)  [0.50, 0.50, 0.50]       (A)   
11   62.04  m_h  -0.646      24    (up)  [0.50, 0.50, 0.50]       (A)   
12   62.04  m_e   0.578      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
13   62.04  m_e   2.013      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
14   62.04  m_e   0.578      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
15   62.04  m_e  16.448      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
16   64.00  m_h  -0.715      23    (up)  [0.50, 0.50, 0.50]       (A)   
17   64.00  m_h  -0.733      23    (up)  [0.50, 0.50, 0.50]       (A)   
18   64.00  m_h  -0.715      24    (up)  [0.50, 0.50, 0.50]       (A)   
19   64.00  m_h  -0.733      24    (up)  [0.50, 0.50, 0.50]       (A)   
20   64.00  m_e   0.592      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
21   64.00  m_e   2.057      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
22   64.00  m_e   0.592      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
23   64.00  m_e  17.454      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
24   66.00  m_h  -0.786      23    (up)  [0.50, 0.50, 0.50]       (A)   
25   66.00  m_h  -0.804      23    (up)  [0.50, 0.50, 0.50]       (A)   
26   66.00  m_h  -0.786      24    (up)  [0.50, 0.50, 0.50]       (A)   
27   66.00  m_h  -0.804      24    (up)  [0.50, 0.50, 0.50]       (A)   
28   66.00  m_e   0.609      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
29   66.00  m_e   2.129      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
..     ...  ...     ...     ...     ...                 ...       ...   
76   78.88  m_h  -3.486      24  (down)  [0.35, 0.00, 0.00]       NaN   
77   78.88  m_h  -2.809      24  (down)  [0.35, 0.00, 0.00]       NaN   
78   78.88  m_e   0.625      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
79   78.88  m_e   2.595      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
80   78.88  m_e   0.625      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
81   78.88  m_e  22.775      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
82   81.17  m_h  -2.661      24    (up)  [0.35, 0.00, 0.00]       NaN   
83   81.17  m_h  -2.511      24    (up)  [0.35, 0.00, 0.00]       NaN   
84   81.17  m_h  -2.661      24  (down)  [0.35, 0.00, 0.00]       NaN   
85   81.17  m_h  -2.511      24  (down)  [0.35, 0.00, 0.00]       NaN   
86   81.17  m_e   0.619      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
87   81.17  m_e   2.700      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
88   81.17  m_e   0.619      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
89   81.17  m_e  22.343      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
90   83.51  m_h  -2.217      24    (up)  [0.35, 0.00, 0.00]       NaN   
91   83.51  m_h  -2.049      24    (up)  [0.35, 0.00, 0.00]       NaN   
92   83.51  m_h  -2.217      24  (down)  [0.35, 0.00, 0.00]       NaN   
93   83.51  m_h  -2.049      24  (down)  [0.35, 0.00, 0.00]       NaN   
94   83.51  m_e   0.613      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
95   83.51  m_e   2.793      25    (up)  [0.00, 0.00, 0.00]   (gamma)   
96   83.51  m_e   0.613      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
97   83.51  m_e  25.055      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
98   85.90  m_h  -1.989      24    (up)  [0.35, 0.00, 0.00]      

la cosa es entonces que para el electron solo tengo que ver todas las masas en gama.
$$
m_x, m_y = m_{\Gamma \rightarrow X}
$$
$$
m_z = m_{\Gamma \rightarrow Z}
$$

## masa del electron

In [26]:
m_e = databands[(databands['mass']=='m_e')]
m_e_gx_up = m_e[(m_e['toname']=='(X)') & (m_e['spin']=='(up)')]
m_e_gx_dn = m_e[(m_e['toname']=='(X)') & (m_e['spin']=='(down)')]
m_e_gz_up = m_e[(m_e['toname']=='(Z)') & (m_e['spin']=='(up)')]
m_e_gz_dn = m_e[(m_e['toname']=='(Z)') & (m_e['spin']=='(down)')]

In [29]:
# guardo las tablas a txts
with open('m_e.dat', 'w') as me:
    me.write('\n=====gamma->X up\n')
m_e_gx_up.to_csv('m_e.dat', sep=' ', index=False, header=True,mode='a')
with open('m_e.dat', 'a') as me:
    me.write('\n=====gamma->X down\n')
m_e_gx_dn.to_csv('m_e.dat', sep=' ', index=False, header=True,mode='a')
with open('m_e.dat', 'a') as me:
    me.write('\n=====gamma-Z Up\n')
m_e_gz_up.to_csv('m_e.dat', sep=' ', index=False, header=True,mode='a')
with open('m_e.dat', 'a') as me:
    me.write('\n=====gamma-Z Down\n')
m_e_gz_dn.to_csv('m_e.dat', sep=' ', index=False, header=True,mode='a')

### electron en el camino $\Gamma \rightarrow Z$, spin down

In [30]:
m_e_gz_dn

Vol mass   value  bandno    spin                from     fname  \
7    60.12  m_e  16.797      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
15   62.04  m_e  16.448      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
23   64.00  m_e  17.454      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
31   66.00  m_e  18.654      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
39   68.04  m_e  18.279      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
49   70.12  m_e  19.707      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
57   72.25  m_e  19.318      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
65   74.41  m_e  21.046      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
73   76.62  m_e  20.639      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
81   78.88  m_e  22.775      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
89   81.17  m_e  22.343      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
97   83.51  m_e  25.055      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
105  85.90  m_e  24.589      25  (down)  [0.00, 0.00, 0.00]   (gamma)   

                     to toname  
7    [0.00, 0.00, 0.50]    (Z)  
15   [0.00, 0.00, 0.50]    (Z)  
23   [0.00, 0.00, 0.50]    (Z)  
31   [0.00, 0.00, 0.50]    (Z)  
39   [0.00, 0.00, 0.50]    (Z)  
49   [0.00, 0.00, 0.50]    (Z)  
57   [0.00, 0.00, 0.50]    (Z)  
65   [0.00, 0.00, 0.50]    (Z)  
73   [0.00, 0.00, 0.50]    (Z)  
81   [0.00, 0.00, 0.50]    (Z)  
89   [0.00, 0.00, 0.50]    (Z)  
97   [0.00, 0.00, 0.50]    (Z)  
105  [0.00, 0.00, 0.50]    (Z)

### spin up

In [31]:
m_e_gz_up

Vol mass  value  bandno  spin                from     fname  \
5    60.12  m_e  1.946      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
13   62.04  m_e  2.013      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
21   64.00  m_e  2.057      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
29   66.00  m_e  2.129      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
37   68.04  m_e  2.200      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
47   70.12  m_e  2.265      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
55   72.25  m_e  2.356      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
63   74.41  m_e  2.441      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
71   76.62  m_e  2.490      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
79   78.88  m_e  2.595      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
87   81.17  m_e  2.700      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
95   83.51  m_e  2.793      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
103  85.90  m_e  2.878      25  (up)  [0.00, 0.00, 0.00]   (gamma)   

                     to toname  
5    [0.00, 0.00, 0.50]    (Z)  
13   [0.00, 0.00, 0.50]    (Z)  
21   [0.00, 0.00, 0.50]    (Z)  
29   [0.00, 0.00, 0.50]    (Z)  
37   [0.00, 0.00, 0.50]    (Z)  
47   [0.00, 0.00, 0.50]    (Z)  
55   [0.00, 0.00, 0.50]    (Z)  
63   [0.00, 0.00, 0.50]    (Z)  
71   [0.00, 0.00, 0.50]    (Z)  
79   [0.00, 0.00, 0.50]    (Z)  
87   [0.00, 0.00, 0.50]    (Z)  
95   [0.00, 0.00, 0.50]    (Z)  
103  [0.00, 0.00, 0.50]    (Z)

notar que en este camino la masa para espin down es 10 veces mayor que con spin down.

## electron camino $\Gamma \rightarrow X$

### spin up

In [32]:
m_e_gx_up

Vol mass  value  bandno  spin                from     fname  \
4    60.12  m_e  0.557      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
12   62.04  m_e  0.578      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
20   64.00  m_e  0.592      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
28   66.00  m_e  0.609      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
36   68.04  m_e  0.620      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
46   70.12  m_e  0.629      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
54   72.25  m_e  0.630      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
62   74.41  m_e  0.635      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
70   76.62  m_e  0.623      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
78   78.88  m_e  0.625      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
86   81.17  m_e  0.619      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
94   83.51  m_e  0.613      25  (up)  [0.00, 0.00, 0.00]   (gamma)   
102  85.90  m_e  0.605      25  (up)  [0.00, 0.00, 0.00]   (gamma)   

                     to toname  
4    [0.50, 0.00, 0.00]    (X)  
12   [0.50, 0.00, 0.00]    (X)  
20   [0.50, 0.00, 0.00]    (X)  
28   [0.50, 0.00, 0.00]    (X)  
36   [0.50, 0.00, 0.00]    (X)  
46   [0.50, 0.00, 0.00]    (X)  
54   [0.50, 0.00, 0.00]    (X)  
62   [0.50, 0.00, 0.00]    (X)  
70   [0.50, 0.00, 0.00]    (X)  
78   [0.50, 0.00, 0.00]    (X)  
86   [0.50, 0.00, 0.00]    (X)  
94   [0.50, 0.00, 0.00]    (X)  
102  [0.50, 0.00, 0.00]    (X)

### spin down

In [33]:
m_e_gx_dn

Vol mass  value  bandno    spin                from     fname  \
6    60.12  m_e  0.557      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
14   62.04  m_e  0.578      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
22   64.00  m_e  0.592      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
30   66.00  m_e  0.609      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
38   68.04  m_e  0.620      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
48   70.12  m_e  0.629      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
56   72.25  m_e  0.630      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
64   74.41  m_e  0.635      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
72   76.62  m_e  0.623      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
80   78.88  m_e  0.625      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
88   81.17  m_e  0.619      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
96   83.51  m_e  0.613      25  (down)  [0.00, 0.00, 0.00]   (gamma)   
104  85.90  m_e  0.605      25  (down)  [0.00, 0.00, 0.00]   (gamma)   

                     to toname  
6    [0.50, 0.00, 0.00]    (X)  
14   [0.50, 0.00, 0.00]    (X)  
22   [0.50, 0.00, 0.00]    (X)  
30   [0.50, 0.00, 0.00]    (X)  
38   [0.50, 0.00, 0.00]    (X)  
48   [0.50, 0.00, 0.00]    (X)  
56   [0.50, 0.00, 0.00]    (X)  
64   [0.50, 0.00, 0.00]    (X)  
72   [0.50, 0.00, 0.00]    (X)  
80   [0.50, 0.00, 0.00]    (X)  
88   [0.50, 0.00, 0.00]    (X)  
96   [0.50, 0.00, 0.00]    (X)  
104  [0.50, 0.00, 0.00]    (X)

En este camino, ambas masas (up y down) son iguales.

### gráficos para electron

# masas del hueco

la masa del hueco tiene un comportamiento mas raro. para empezar, esta desdoblada en dos bandas segun la direccion. 

In [29]:
m_h = databands[(databands['mass']=='m_h')]
# m_h = databands[(databands['fname']=='(gamma)') & (databands['toname']=='(X)') & (databands['spin']=='(up)')]

In [45]:
m_h_za = databands[(databands['mass']=='m_h') & (databands['toname']=='(A)')]
m_h_zr = databands[(databands['mass']=='m_h') & (databands['toname']=='(R)')]
m_h_tz = databands[(databands['mass']=='m_h') & (databands['toname']=='(Z)')]
m_h_fz = databands[(databands['mass']=='m_h') & (databands['fname']==' (Z)')]

In [32]:
m_h

Vol mass  value  bandno    spin                from fname  \
0    60.12  m_h -0.588      23    (up)  [0.50, 0.50, 0.50]   (A)   
1    60.12  m_h -0.605      23    (up)  [0.50, 0.50, 0.50]   (A)   
2    60.12  m_h -0.588      24    (up)  [0.50, 0.50, 0.50]   (A)   
3    60.12  m_h -0.605      24    (up)  [0.50, 0.50, 0.50]   (A)   
8    62.04  m_h -0.628      23    (up)  [0.50, 0.50, 0.50]   (A)   
9    62.04  m_h -0.646      23    (up)  [0.50, 0.50, 0.50]   (A)   
10   62.04  m_h -0.628      24    (up)  [0.50, 0.50, 0.50]   (A)   
11   62.04  m_h -0.646      24    (up)  [0.50, 0.50, 0.50]   (A)   
16   64.00  m_h -0.715      23    (up)  [0.50, 0.50, 0.50]   (A)   
17   64.00  m_h -0.733      23    (up)  [0.50, 0.50, 0.50]   (A)   
18   64.00  m_h -0.715      24    (up)  [0.50, 0.50, 0.50]   (A)   
19   64.00  m_h -0.733      24    (up)  [0.50, 0.50, 0.50]   (A)   
24   66.00  m_h -0.786      23    (up)  [0.50, 0.50, 0.50]   (A)   
25   66.00  m_h -0.804      23    (up)  [0.50, 0.50, 0.50]   (A)   
26   66.00  m_h -0.786      24    (up)  [0.50, 0.50, 0.50]   (A)   
27   66.00  m_h -0.804      24    (up)  [0.50, 0.50, 0.50]   (A)   
32   68.04  m_h -0.899      23    (up)  [0.50, 0.50, 0.50]   (A)   
33   68.04  m_h -0.919      23    (up)  [0.50, 0.50, 0.50]   (A)   
34   68.04  m_h -0.899      24    (up)  [0.50, 0.50, 0.50]   (A)   
35   68.04  m_h -0.919      24    (up)  [0.50, 0.50, 0.50]   (A)   
40   70.12  m_h -7.820      23    (up)  [0.00, 0.00, 0.50]   (Z)   
41   70.12  m_h -0.715      23    (up)  [0.00, 0.00, 0.50]   (Z)   
42   70.12  m_h -1.023      23    (up)  [0.00, 0.00, 0.50]   (Z)   
43   70.12  m_h -7.820      24    (up)  [0.00, 0.00, 0.50]   (Z)   
44   70.12  m_h -1.301      24    (up)  [0.00, 0.00, 0.50]   (Z)   
45   70.12  m_h -1.023      24    (up)  [0.00, 0.00, 0.50]   (Z)   
50   72.25  m_h -4.311      24    (up)  [0.31, 0.00, 0.00]   NaN   
51   72.25  m_h -6.054      24    (up)  [0.31, 0.00, 0.00]   NaN   
52   72.25  m_h -4.311      24  (down)  [0.31, 0.00, 0.00]   NaN   
53   72.25  m_h -6.054      24  (down)  [0.31, 0.00, 0.00]   NaN   
58   74.41  m_h -5.110      24    (up)  [0.33, 0.00, 0.00]   NaN   
59   74.41  m_h -4.245      24    (up)  [0.33, 0.00, 0.00]   NaN   
60   74.41  m_h -5.110      24  (down)  [0.33, 0.00, 0.00]   NaN   
61   74.41  m_h -4.245      24  (down)  [0.33, 0.00, 0.00]   NaN   
66   76.62  m_h -4.696      24    (up)  [0.35, 0.00, 0.00]   NaN   
67   76.62  m_h -2.826      24    (up)  [0.35, 0.00, 0.00]   NaN   
68   76.62  m_h -4.696      24  (down)  [0.35, 0.00, 0.00]   NaN   
69   76.62  m_h -2.826      24  (down)  [0.35, 0.00, 0.00]   NaN   
74   78.88  m_h -3.486      24    (up)  [0.35, 0.00, 0.00]   NaN   
75   78.88  m_h -2.809      24    (up)  [0.35, 0.00, 0.00]   NaN   
76   78.88  m_h -3.486      24  (down)  [0.35, 0.00, 0.00]   NaN   
77   78.88  m_h -2.809      24  (down)  [0.35, 0.00, 0.00]   NaN   
82   81.17  m_h -2.661      24    (up)  [0.35, 0.00, 0.00]   NaN   
83   81.17  m_h -2.511      24    (up)  [0.35, 0.00, 0.00]   NaN   
84   81.17  m_h -2.661      24  (down)  [0.35, 0.00, 0.00]   NaN   
85   81.17  m_h -2.511      24  (down)  [0.35, 0.00, 0.00]   NaN   
90   83.51  m_h -2.217      24    (up)  [0.35, 0.00, 0.00]   NaN   
91   83.51  m_h -2.049      24    (up)  [0.35, 0.00, 0.00]   NaN   
92   83.51  m_h -2.217      24  (down)  [0.35, 0.00, 0.00]   NaN   
93   83.51  m_h -2.049      24  (down)  [0.35, 0.00, 0.00]   NaN   
98   85.90  m_h -1.989      24    (up)  [0.35, 0.00, 0.00]   NaN   
99   85.90  m_h -1.589      24    (up)  [0.35, 0.00, 0.00]   NaN   
100  85.90  m_h -1.989      24  (down)  [0.35, 0.00, 0.00]   NaN   
101  85.90  m_h -1.589      24  (down)  [0.35, 0.00, 0.00]   NaN   

                     to   toname  
0    [0.00, 0.50, 0.50]      (R)  
1    [0.00, 0.00, 0.50]      (Z)  
2    [0.00, 0.50, 0.50]      (R)  
3    [0.00, 0.00, 0.50]      (Z)  
8    [0.00, 0.50, 0.50]      (R)  
9    [0.00, 0.00, 0.50]      (Z)  
10   [0.00, 0.50, 0.5

## Hueco en el camino $Z \rightarrow A$

In [31]:
m_h_za

Vol mass  value  bandno  spin                from fname  \
42  70.12  m_h -1.023      23  (up)  [0.00, 0.00, 0.50]   (Z)   
45  70.12  m_h -1.023      24  (up)  [0.00, 0.00, 0.50]   (Z)   

                    to toname  
42  [0.50, 0.50, 0.50]    (A)  
45  [0.50, 0.50, 0.50]    (A)

### En el camino $A \rightarrow R$

In [327]:
m_h_zr

Vol mass  value  bandno  spin                from fname  \
0   60.12  m_h -0.588      23  (up)  [0.50, 0.50, 0.50]   (A)   
2   60.12  m_h -0.588      24  (up)  [0.50, 0.50, 0.50]   (A)   
8   62.04  m_h -0.628      23  (up)  [0.50, 0.50, 0.50]   (A)   
10  62.04  m_h -0.628      24  (up)  [0.50, 0.50, 0.50]   (A)   
16  64.00  m_h -0.715      23  (up)  [0.50, 0.50, 0.50]   (A)   
18  64.00  m_h -0.715      24  (up)  [0.50, 0.50, 0.50]   (A)   
24  66.00  m_h -0.786      23  (up)  [0.50, 0.50, 0.50]   (A)   
26  66.00  m_h -0.786      24  (up)  [0.50, 0.50, 0.50]   (A)   
32  68.04  m_h -0.899      23  (up)  [0.50, 0.50, 0.50]   (A)   
34  68.04  m_h -0.899      24  (up)  [0.50, 0.50, 0.50]   (A)   
41  70.12  m_h -0.715      23  (up)  [0.00, 0.00, 0.50]   (Z)   
44  70.12  m_h -1.301      24  (up)  [0.00, 0.00, 0.50]   (Z)   

                    to toname  
0   [0.00, 0.50, 0.50]    (R)  
2   [0.00, 0.50, 0.50]    (R)  
8   [0.00, 0.50, 0.50]    (R)  
10  [0.00, 0.50, 0.50]    (R)  
16  [0.00, 0.50, 0.50]    (R)  
18  [0.00, 0.50, 0.50]    (R)  
24  [0.00, 0.50, 0.50]    (R)  
26  [0.00, 0.50, 0.50]    (R)  
32  [0.00, 0.50, 0.50]    (R)  
34  [0.00, 0.50, 0.50]    (R)  
41  [0.00, 0.50, 0.50]    (R)  
44  [0.00, 0.50, 0.50]    (R)

In [328]:
m_h_tz

Vol mass  value  bandno  spin                from fname  \
1   60.12  m_h -0.605      23  (up)  [0.50, 0.50, 0.50]   (A)   
3   60.12  m_h -0.605      24  (up)  [0.50, 0.50, 0.50]   (A)   
9   62.04  m_h -0.646      23  (up)  [0.50, 0.50, 0.50]   (A)   
11  62.04  m_h -0.646      24  (up)  [0.50, 0.50, 0.50]   (A)   
17  64.00  m_h -0.733      23  (up)  [0.50, 0.50, 0.50]   (A)   
19  64.00  m_h -0.733      24  (up)  [0.50, 0.50, 0.50]   (A)   
25  66.00  m_h -0.804      23  (up)  [0.50, 0.50, 0.50]   (A)   
27  66.00  m_h -0.804      24  (up)  [0.50, 0.50, 0.50]   (A)   
33  68.04  m_h -0.919      23  (up)  [0.50, 0.50, 0.50]   (A)   
35  68.04  m_h -0.919      24  (up)  [0.50, 0.50, 0.50]   (A)   

                    to toname  
1   [0.00, 0.00, 0.50]    (Z)  
3   [0.00, 0.00, 0.50]    (Z)  
9   [0.00, 0.00, 0.50]    (Z)  
11  [0.00, 0.00, 0.50]    (Z)  
17  [0.00, 0.00, 0.50]    (Z)  
19  [0.00, 0.00, 0.50]    (Z)  
25  [0.00, 0.00, 0.50]    (Z)  
27  [0.00, 0.00, 0.50]    (Z)  
33  [0.00, 0.00, 0.50]    (Z)  
35  [0.00, 0.00, 0.50]    (Z)

In [331]:
m_h_fz

Vol mass  value  bandno  spin                from fname  \
40  70.12  m_h -7.820      23  (up)  [0.00, 0.00, 0.50]   (Z)   
41  70.12  m_h -0.715      23  (up)  [0.00, 0.00, 0.50]   (Z)   
42  70.12  m_h -1.023      23  (up)  [0.00, 0.00, 0.50]   (Z)   
43  70.12  m_h -7.820      24  (up)  [0.00, 0.00, 0.50]   (Z)   
44  70.12  m_h -1.301      24  (up)  [0.00, 0.00, 0.50]   (Z)   
45  70.12  m_h -1.023      24  (up)  [0.00, 0.00, 0.50]   (Z)   

                    to   toname  
40  [0.00, 0.00, 0.00]  (gamma)  
41  [0.00, 0.50, 0.50]      (R)  
42  [0.50, 0.50, 0.50]      (A)  
43  [0.00, 0.00, 0.00]  (gamma)  
44  [0.00, 0.50, 0.50]      (R)  
45  [0.50, 0.50, 0.50]      (A)

todos los valores que puse ahi estan $v < v_{eq}$, que pasa arriva ?

In [53]:
m_h_vbig = databands[(databands['Vol']>vopt) & (databands['mass']=='m_h')]

In [335]:
m_h_vbig

Vol mass  value  bandno    spin                from fname  \
50   72.25  m_h -4.311      24    (up)  [0.31, 0.00, 0.00]   NaN   
51   72.25  m_h -6.054      24    (up)  [0.31, 0.00, 0.00]   NaN   
52   72.25  m_h -4.311      24  (down)  [0.31, 0.00, 0.00]   NaN   
53   72.25  m_h -6.054      24  (down)  [0.31, 0.00, 0.00]   NaN   
58   74.41  m_h -5.110      24    (up)  [0.33, 0.00, 0.00]   NaN   
59   74.41  m_h -4.245      24    (up)  [0.33, 0.00, 0.00]   NaN   
60   74.41  m_h -5.110      24  (down)  [0.33, 0.00, 0.00]   NaN   
61   74.41  m_h -4.245      24  (down)  [0.33, 0.00, 0.00]   NaN   
66   76.62  m_h -4.696      24    (up)  [0.35, 0.00, 0.00]   NaN   
67   76.62  m_h -2.826      24    (up)  [0.35, 0.00, 0.00]   NaN   
68   76.62  m_h -4.696      24  (down)  [0.35, 0.00, 0.00]   NaN   
69   76.62  m_h -2.826      24  (down)  [0.35, 0.00, 0.00]   NaN   
74   78.88  m_h -3.486      24    (up)  [0.35, 0.00, 0.00]   NaN   
75   78.88  m_h -2.809      24    (up)  [0.35, 0.00, 0.00]   NaN   
76   78.88  m_h -3.486      24  (down)  [0.35, 0.00, 0.00]   NaN   
77   78.88  m_h -2.809      24  (down)  [0.35, 0.00, 0.00]   NaN   
82   81.17  m_h -2.661      24    (up)  [0.35, 0.00, 0.00]   NaN   
83   81.17  m_h -2.511      24    (up)  [0.35, 0.00, 0.00]   NaN   
84   81.17  m_h -2.661      24  (down)  [0.35, 0.00, 0.00]   NaN   
85   81.17  m_h -2.511      24  (down)  [0.35, 0.00, 0.00]   NaN   
90   83.51  m_h -2.217      24    (up)  [0.35, 0.00, 0.00]   NaN   
91   83.51  m_h -2.049      24    (up)  [0.35, 0.00, 0.00]   NaN   
92   83.51  m_h -2.217      24  (down)  [0.35, 0.00, 0.00]   NaN   
93   83.51  m_h -2.049      24  (down)  [0.35, 0.00, 0.00]   NaN   
98   85.90  m_h -1.989      24    (up)  [0.35, 0.00, 0.00]   NaN   
99   85.90  m_h -1.589      24    (up)  [0.35, 0.00, 0.00]   NaN   
100  85.90  m_h -1.989      24  (down)  [0.35, 0.00, 0.00]   NaN   
101  85.90  m_h -1.589      24  (down)  [0.35, 0.00, 0.00]   NaN   

                     to   toname  
50   [0.00, 0.00, 0.00]  (gamma)  
51   [0.50, 0.00, 0.00]      (X)  
52   [0.00, 0.00, 0.00]  (gamma)  
53   [0.50, 0.00, 0.00]      (X)  
58   [0.00, 0.00, 0.00]  (gamma)  
59   [0.50, 0.00, 0.00]      (X)  
60   [0.00, 0.00, 0.00]  (gamma)  
61   [0.50, 0.00, 0.00]      (X)  
66   [0.00, 0.00, 0.00]  (gamma)  
67   [0.50, 0.00, 0.00]      (X)  
68   [0.00, 0.00, 0.00]  (gamma)  
69   [0.50, 0.00, 0.00]      (X)  
74   [0.00, 0.00, 0.00]  (gamma)  
75   [0.50, 0.00, 0.00]      (X)  
76   [0.00, 0.00, 0.00]  (gamma)  
77   [0.50, 0.00, 0.00]      (X)  
82   [0.00, 0.00, 0.00]  (gamma)  
83   [0.50, 0.00, 0.00]      (X)  
84   [0.00, 0.00, 0.00]  (gamma)  
85   [0.50, 0.00, 0.00]      (X)  
90   [0.00, 0.00, 0.00]  (gamma)  
91   [0.50, 0.00, 0.00]      (X)  
92   [0.00, 0.00, 0.00]  (gamma)  
93   [0.50, 0.00, 0.00]      (X)  
98   [0.00, 0.00, 0.00]  (gamma)  
99   [0.50, 0.00, 0.00]      (X)  
100  [0.00, 0.00, 0.00]  (gamma)  
101  [0.50, 0.00, 0.00]      (X)

es curioso que para volumenes grandes, el máximo de la banda de valencia se corre de $Z$ a $[0.33, 0, 0 ]$ !